scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -32,23 +32,43 @@
   ; suppress otherwise identical states so we don't spam.
   (= old-state new-state))
 
-(defn remove-finalization-if-set-and-delete
-  [api-client expected-state-dict pod]
-  (api/remove-finalization-if-set-and-delete api-client pod)
-  expected-state-dict)
+(defn delete-task
+  "Kill task is the same as deleting a task. I semantically distinguish them. Delete is used for completed tasks that
+  we're done with. Kill is used for possibly running tasks we want to kill so that they fail. Returns a new expected state
+  dict of nil."
+  [api-client pod]
+  (api/kill-task api-client pod)
```

Comment:
Is this killing a task or killing the pod? I think it should be named `kill-pod` to make it clearer.

Reply:
Can we rename `api/kill-task` to `api/kill-pod` as that is what it is appearing to do?

Reply:
I want to call if with why its called, not what it does.

Reply:
Is it killing a cook job _instance_? If so, then the name should reflect instance instead of task?